pydantic/_internal/_fields.py

```diff
@@ -277,3 +213,45 @@ def _is_finalvar_with_default_val(type_: type[Any], val: Any) -> bool:
         return False
     else:
         return True
+
+
+def collect_dataclass_fields(
+    cls: type[StandardDataclass], types_namespace: dict[str, Any] | None, *, typevars_map: dict[Any, Any] | None = None
+) -> dict[str, FieldInfo]:
+    from ..fields import FieldInfo
+
+    fields: dict[str, FieldInfo] = {}
+    type_hints = get_cls_type_hints_lenient(cls, types_namespace)
+    dataclass_fields: dict[str, dataclasses.Field] = cls.__dataclass_fields__
+
+    for ann_name, dataclass_field in dataclass_fields.items():
```

Comment:
The trick to simplifying here is to use cls.__dataclass_fields__ to do a lot of the heavy lifting.

Note I still kept the `get_cls_type_hints_lenient` call around to deal with the fact that the `dataclasses.Field` will not resolve forward references/strings.

Reply:
done (will be in next pushed commit)